community/modules/compute/schedmd-slurm-gcp-v5-node-group/source_image_logic.tf

```diff
@@ -24,3 +30,27 @@ locals {
     : "projects/${local.source_image_project}/global/images/family"
   )
 }
+
+data "google_compute_image" "slurm" {
+  family  = contains(keys(var.instance_image), "family") ? var.instance_image["family"] : null
+  name    = contains(keys(var.instance_image), "name") ? var.instance_image["name"] : null
+  project = var.instance_image["project"]
+
+  lifecycle {
```

Comment:
I believe that we want to distinguish between 2 situations:

1. The blueprint is using the schedmd-slurm-public project but an incompatible image.
2. The blueprint is using another project entirely

I'm also concerned about throwing the words "not supported" around when we really mean "incompatible." Let's discuss specific wording offline.

I think you'll want two postconditions for this and it will probably look something like below. The error messages are not intended to be taken verbatim, but an idea of how to distinguish between the different scenarios.

```
  condition =  var.instance_image_override || self.project == "schedmd-slurm-public"
  error_message = "If using images published outside the schedmd-slurm-public project, you must set var.instance_image_override to true and build images according to LINK-TO-URL."
```

The OR condition could also be a `contains(keys...`. A second postcondition might look like

```
  condition =  self.project != "schedmd-slurm-public" || try(contains(local.supported_project_families[self.project], self.family), false)
  error_message = "If using images published in the schedmd-slurm-public project, you must use one of these compatible releases: X,Y,Z"
```

I don't think we want to get in the business of supporting alternative families within the schedmd project.

Reply:
I've updated the language (but we should still discuss), and took a different approach for the two potential errors.  Instead of two postconditions, which may both trigger, I've created one postcondition with a conditional for the error message.  It should give a more accurate error message depending on if the user has specified an incorrect project or image family.

Reply:
I do not see how my two postconditions can simultaneously trigger. Can you demonstrate a combination of settings that does?

My suggestion seems very direct to read and the error messages are a lot shorter. I want to get these error messages really down to "Here is your immediate problem. Visit URL for details." Preferably one line.

Reply:
Resolved.